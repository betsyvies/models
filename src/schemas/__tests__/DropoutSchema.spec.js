const mongoose = require('mongoose/browser');
const DropoutSchema = require('../DropoutSchema')(mongoose);

describe('DropoutSchema', () => {
  it('should fail validation when fields are missing', () => {
    const doc = new mongoose.Document({}, DropoutSchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });
  it('should successfully validate with proper values', (done) => {
    const doc = new mongoose.Document({
      city: 'LIM',
      cohort: 'LIM007',
      fullName: 'Diego Vélez',
      email: 'someone@somewhere.com',
      signUpCohortCity: 'BOG',
      date: '5/14/2018',
      stage: 'Project 1',
      studentCode: 'LIM181080',
      reason: 'Leave the program voluntarily',
      reasonDetail: 'Problemas familiares y económicos',
      notes: 'Una lástima, era una buena estudiante :(',
      sad: true,
      covidRelated: true,
    }, DropoutSchema);
    return doc.validate((err) => {
      expect(err).toBe(null);
      done();
    });
  });
});
