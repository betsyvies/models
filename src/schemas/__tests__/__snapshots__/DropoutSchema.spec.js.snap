// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropoutSchema should fail validation when fields are missing 1`] = `
Object {
  "city": [ValidatorError: Path \`city\` is required.],
  "cohort": [ValidatorError: Path \`cohort\` is required.],
  "covidRelated": [ValidatorError: Path \`covidRelated\` is required.],
  "email": [ValidatorError: Path \`email\` is required.],
  "fullName": [ValidatorError: Path \`fullName\` is required.],
  "notes": [ValidatorError: Path \`notes\` is required.],
  "reason": [ValidatorError: Path \`reason\` is required.],
  "reasonDetail": [ValidatorError: Path \`reasonDetail\` is required.],
  "sad": [ValidatorError: Path \`sad\` is required.],
  "signUpCohortCity": [ValidatorError: Path \`signUpCohortCity\` is required.],
  "stage": [ValidatorError: Path \`stage\` is required.],
  "studentCode": [ValidatorError: Path \`studentCode\` is required.],
}
`;
